numpy/ma/core.py

```diff
@@ -5869,6 +5869,18 @@ def __getitem__(self, indx):
 
         """
         m = self._mask
+        if m[indx].size > 1:
```

Comment:
It is possible to have subarrays of size 1, eg `dtype(["A", "i2", (1,1,1)])` so this probably needs something better. `m[indx].shape == ()` was an idea, but that won't work for object types. Probably you just have to check `isinstance(m[indx], np.ndarray)`.